transfer/dummy.py

```diff
@@ -23,24 +30,39 @@ def main():
         test_data = test_data.reshape(test_data.shape[0], MNIST_IMAGE_SIZE, MNIST_IMAGE_SIZE, 1)
         input_shape = (MNIST_IMAGE_SIZE, MNIST_IMAGE_SIZE, 1)
 
+
+    #heuristic bias
     train_data = train_data * fac + 0.01
     test_data = test_data * fac + 0.01
+
+    #takes an array (n1,n2,...n_k)
+    #so that n_i in {0,1,...t}
+    #returns a matrix in the standard basis
+    # (,,,3,,,) -> (001000...)
     train_labels = tf.keras.utils.to_categorical(train_labels, num_classes=NUMBER_OF_LABELS)
     test_labels = tf.keras.utils.to_categorical(test_labels, num_classes=NUMBER_OF_LABELS)
 
+
     model = tf.keras.models.Sequential()
-    model.add(tf.keras.layers.Conv2D(filters=2, kernel_size=(3, 3), activation='relu', input_shape=input_shape))
-    model.add(tf.keras.layers.MaxPooling2D())
-    model.add(tf.keras.layers.Conv2D(filters=2, kernel_size=(3, 3), activation='relu'))
-    model.add(tf.keras.layers.Flatten())
-    model.add(tf.keras.layers.Dense(activation='relu', units=50))
-    model.add(tf.keras.layers.Dense(activation='relu', units=30))
-    model.add(tf.keras.layers.Dense(activation='softmax', units=10))
-
-    model.compile(loss=tf.keras.losses.categorical_crossentropy,
-                  optimizer=tf.keras.optimizers.Adadelta(), metrics=['accuracy'])
-    history = model.fit(x=train_data, y=train_labels, batch_size=10, epochs=8, verbose=1)
 
+    model.add(tf.keras.models.Conv2D(32, kernel_size=(3, 3), activation='relu',
+                     input_shape=input_shape))
+
+    model.add(tf.keras.models.Conv2D(64, (3, 3), activation='relu'))
+
+    # reduce dim, choose best
+    model.add(tf.keras.models.MaxPooling2D(pool_size=(2, 2)))
+
+    #model.add(Dropout(0.25))
+    model.add(tf.keras.models.Flatten())
+
+    model.add(tf.keras.models.Dense(128, activation='relu'))
+
+    #model.add(Dropout(0.5))
+    model.add(tf.keras.models.Dense(NUMBER_OF_LABELS, activation='softmax'))
+    model.compile(loss=tf.keras.losses.categorical_crossentropy, optimizer=tf.keras.optimizers.SGD, metrics=['accuracy'])
+
+    history = model.fit(x=train_data, y=train_labels, batch_size=10, epochs=16, verbose=1)
     result = model.predict(test_data[0:10])
```

Comment:
you can delete the result print, we draw a graph instead

Reply:
removed printing